corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js

```diff
@@ -124,10 +134,16 @@ hqDefine("cloudcare/js/formplayer/menus/collections", [
                 // backwards compatibility - remove after FP deploy of #1374
                 _.defaults(response, {"hasDetails": true});
                 _.extend(this, _.pick(response, this.entityProperties));
+                if (response.locales && !UsersModels.getCurrentUser().displayOptions.singleAppMode) {
+                    enableLanguageOption(response, this);
+                }
                 return response.entities;
             } else if (response.type === "query") {
                 addBreadcrumb(this, "query", sentryData);
                 _.extend(this, _.pick(response, this.queryProperties));
+                if (response.locales && !UsersModels.getCurrentUser().displayOptions.singleAppMode) {
+                    enableLanguageOption(response, this);
+                }
```

Comment:
I think this would make sense to do outside this `if/else` block, would that be possible? Or does it depend on the state changed by the `_.extend` calls?

Reply:
sure, I'd just need to have one more condition like `&& ["query", "entities"].includes(response.type)`